pageserver/src/context.rs

```diff
@@ -88,13 +88,16 @@
 
 use crate::task_mgr::TaskKind;
 
+pub(crate) mod optional_counter;
+
 // The main structure of this module, see module-level comment.
-#[derive(Clone, Debug)]
+#[derive(Debug)]
 pub struct RequestContext {
     task_kind: TaskKind,
     download_behavior: DownloadBehavior,
     access_stats_behavior: AccessStatsBehavior,
     page_content_kind: PageContentKind,
+    pub micros_spent_throttled: optional_counter::MicroSecondsCounterU32,
```

Comment:
future: the atomic in here makes me think that maybe we should use `&mut RequestContext` so that we can use a non-atomic value here -- when doing ordinary request handling that doesn't jump between tasks (I haven't checked, but I'm guessing getpage requests would fit that category)

Reply:
Hm yeah, I guess passing a `&mut RequestContext` instead of a `&RequestContext` wouldn't hurt, we could try to convert the code base to that in another PR.

Related: mid-term, I want to move away of passing a copy of RequestContext to child taks, and instead do something like `struct RequestContext { inner: Arc<Inner> }` , so we can form a tree. Maybe we can even go crazy and do `trait RequestContext` instead so a given susbystem can extend the context via downcasts. (Basically I just want Go's context.Context, maybe with stronger typing).